lib/stormpath-sdk/resource/directory.rb

```diff
@@ -46,17 +46,27 @@ def provider
   end
 
   def provider_metadata
-    metadata_href = provider.service_provider_metadata["href"]
+    metadata_href = provider.service_provider_metadata['href']
     data_store.get_resource metadata_href, Stormpath::Provider::SamlProviderMetadata
   end
 
   def statement_mapping_rules
-    metadata_href = provider.attribute_statement_mapping_rules["href"]
+    metadata_href = provider.attribute_statement_mapping_rules['href']
     data_store.get_resource metadata_href, Stormpath::Provider::SamlMappingRules
   end
 
   def create_attribute_mappings(mappings)
-    mappings.set_options(href: provider.attribute_statement_mapping_rules["href"])
+    mappings.set_options(href: provider.attribute_statement_mapping_rules['href'])
     data_store.create mappings.href, mappings, Stormpath::Provider::SamlMappingRules
   end
+
+  def user_info_mapping_rules
+    metadata_href = provider.user_info_mapping_rules['href']
```

Comment:
this is not needed. Eventually delegate the `user_info_mapping_rules` to the provider.